src/fs_event_wrap.cc

```diff
@@ -156,7 +160,31 @@ void FSEventWrap::OnEvent(uv_fs_event_t* handle, const char* filename,
   };
 
   if (filename != nullptr) {
-    argv[2] = OneByteString(env->isolate(), filename);
+    /* If encoding is BUFFER, return a Buffer.
+       If encoding is UCS2, attempt to decode as UCS2 or report an error.
+       Otherwise, Encode as the requested type, with UTF8 as the fallback. */
+    if (wrap->encoding_ == BUFFER) {
+      argv[2] = Buffer::Copy(env,
+                             filename,
+                             strlen(filename)).ToLocalChecked().As<Value>();
+    } else if (wrap->encoding_ == UCS2) {
+      if (strlen(filename) % 2 != 0) {
+        argv[0] = Integer::New(env->isolate(), UV__ECHARSET);
+        argv[2] = StringBytes::Encode(env->isolate(),
+                                      filename,
+                                      strlen(filename),
+                                      BINARY);
+      } else {
+        argv[2] = StringBytes::Encode(env->isolate(),
+            reinterpret_cast<const uint16_t*>(filename),
```

Comment:
@bnoordhuis how would we deal with the case where user says the filename is encoded in utf16le?